lib/src/room.dart

```diff
@@ -1648,102 +1650,171 @@ class Room {
     }
   }
 
-  final Set<String> _requestingMatrixIds = {};
-
-  /// Requests a missing [User] for this room. Important for clients using
-  /// lazy loading. If the user can't be found this method tries to fetch
-  /// the displayname and avatar from the profile if [requestProfile] is true.
-  Future<User?> requestUser(
+  // Internal helper to implement requestUser
+  Future<User?> _requestSingleParticipantViaState(
     String mxID, {
-    bool ignoreErrors = false,
-    bool requestProfile = true,
+    required bool ignoreErrors,
   }) async {
-    assert(mxID.isValidMatrixId);
+    try {
+      Logs().v(
+        'Request missing user $mxID in room ${getLocalizedDisplayname()} from the server...',
+      );
+      final resp = await client.getRoomStateWithKey(
+        id,
+        EventTypes.RoomMember,
+        mxID,
+      );
 
-    // Is user already in cache?
-    var foundUser = getState(EventTypes.RoomMember, mxID)?.asUser(this);
+      // valid member events require a valid membership key
+      final membership = resp.tryGet<String>('membership', TryGet.required);
+      assert(membership != null);
```

Comment:
Why assert here?

Reply:
Because our tests actually had that field unset. Such events should never exist, but we might generate them internally, so this just makes sure such an event shows up as an error in tests and during development. Basically the assertion checks, that our asumption is correct. That's also what the comment explains.